ceremonies/src/lib.rs

```diff
@@ -443,11 +443,17 @@ pub mod pallet {
 		) -> DispatchResultWithPostInfo {
 			let participant = &ensure_signed(origin)?;
 
-			if <encointer_scheduler::Pallet<T>>::current_phase() != CeremonyPhaseType::Registering {
+			let current_phase = <encointer_scheduler::Pallet<T>>::current_phase();
+			if ![CeremonyPhaseType::Registering, CeremonyPhaseType::Attesting]
+				.contains(&current_phase)
+			{
 				return Err(<Error<T>>::WrongPhaseForClaimingRewards.into())
 			}
 
-			let cindex = <encointer_scheduler::Pallet<T>>::current_ceremony_index() - 1; //safe; cindex comes from within
+			let cindex_offset = if current_phase == CeremonyPhaseType::Registering { 1 } else { 0 };
```

Comment:
Detail: what about a match statement on the current_phase which solves the ensure Phase and the offset at the same time? might be more readable?